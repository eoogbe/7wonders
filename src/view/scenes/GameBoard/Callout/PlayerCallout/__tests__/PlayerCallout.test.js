/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import PLAYERS from '__fixtures__/players';
import PlayerCallout from '../PlayerCallout';

const PLAYER_ID = 'PLAYER2';

const mockStore = configureStore();

it('renders without crashing', () => {
  const store = mockStore(createState());
  const tree =
    renderer.create(
      <Provider store={store}>
        <PlayerCallout playerId={PLAYER_ID} />
      </Provider>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});

const createState = () => ({
  players: PLAYERS,
});
