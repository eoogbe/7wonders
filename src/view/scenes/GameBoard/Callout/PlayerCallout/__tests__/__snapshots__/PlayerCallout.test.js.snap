// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="Tableau"
  style={Object {}}
>
  <ul
    className="Tableau-cardList"
  >
    <li
      className="TableauCard"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": "rgb(141,110,99)",
          "border": "1px solid rgba(93, 64, 55)",
        }
      }
    >
      <ul
        className="TableauCard-effectList"
      >
        <li
          className="TableauCard-effect"
        >
          Ore
        </li>
      </ul>
    </li>
    <li
      className="TableauCard"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": "rgb(141,110,99)",
          "border": "1px solid rgba(93, 64, 55)",
        }
      }
    >
      <ul
        className="TableauCard-effectList"
      >
        <li
          className="TableauCard-effect"
        >
          Clay / Ore
        </li>
      </ul>
    </li>
  </ul>
  <ul
    className="Tableau-cardList"
  >
    <li
      className="TableauCard"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": "rgb(33,150,243)",
          "border": "1px solid rgba(93, 64, 55)",
        }
      }
    >
      <ul
        className="TableauCard-effectList"
      >
        <li
          className="TableauCard-effect"
        >
          3 points
        </li>
      </ul>
    </li>
    <li
      className="TableauCard"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": "rgb(33,150,243)",
          "border": "1px solid rgba(93, 64, 55)",
        }
      }
    >
      <ul
        className="TableauCard-effectList"
      >
        <li
          className="TableauCard-effect"
        >
          2 points
        </li>
      </ul>
    </li>
  </ul>
</div>
`;
